tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -129,14 +131,14 @@ export class WebGPUBackend extends KernelBackend {
   private dummyContext: GPUCanvasContext;
   private tensorDataPendingDisposal: DataId[] = [];
   private static nextDataId = 0;
-  private pipelineCache:
-      {[key: string]: GPUComputePipeline|Promise<GPUComputePipeline>};
+  private pipelineOrShaderCache: {[key: string]: PipelineOrShader};
```

Comment:
Is it better to define `pipelineCache` as `{[key: string]: {pipeline?: GPUComputePipeline|Promise<GPUComputePipeline>; source?: string;}`.
I don't like the value must be either pipeline or shader. They can't be coexisted.

In another side, is it necessary to cache the shader source? It seems like single use. It's not big deal to dynamically generate it when needed.

Reply:
I split these to two, PTAL